cmd/podman/common/specgen.go

```diff
@@ -656,25 +655,52 @@ func FillOutSpecGen(s *specgen.SpecGenerator, c *ContainerCLIOpts, args []string
 }
 
 func makeHealthCheckFromCli(inCmd, interval string, retries uint, timeout, startPeriod string) (*manifest.Schema2HealthConfig, error) {
+	cmdArr := []string{}
+	isArr := false
+	cmdSplit := strings.SplitN(inCmd, " ", 2) // default for compat
+	if strings.Contains(inCmd, "[") {         // given an array, need to unmarshall it into a pointer
+		isArr = true
+		err := json.Unmarshal([]byte(inCmd), &cmdArr)
+		if err != nil {
+			return nil, err
+		}
+		cmdSplit = cmdArr
+	}
+	concat := ""
+	if cmdSplit[0] != "CMD" && cmdSplit[0] != "none" && cmdSplit[0] != "CMD-SHELL" { // if it isnt a cmd-shell or a cmd, it is a command
```

Comment:
What if the result is len(0)? This will panic

Reply:
an empty string does not panic and shows no healthcheck on inspect since no cmd was specified